gimbal-adapter/src/main/java/com/urbanairship/gimbal/GimbalAdapter.java

```diff
@@ -350,43 +349,48 @@ public boolean isPermissionGranted() {
         return false;
     }
 
+    /**
+     * Called when the airship channel is created.
+     */
+    public void onAirshipChannelCreated() {
+        if (isStarted()) {
+            updateDeviceAttributes();
+        }
+    }
 
     /**
      * Updates Gimbal and Urban Airship device attributes.
      */
-    void updateDeviceAttributes() {
-        String gimbalApiKey = preferences.getString(API_KEY_PREFERENCE, null);
-        if (gimbalApiKey != null) {
+    private void updateDeviceAttributes() {
+        Map<String, String> deviceAttributes = new HashMap<>();
 
-            Map<String, String> deviceAttributes = new HashMap<>();
-
-            if (DeviceAttributesManager.getInstance().getDeviceAttributes() != null
-                    && DeviceAttributesManager.getInstance().getDeviceAttributes().size() > 0) {
-                deviceAttributes.putAll(DeviceAttributesManager.getInstance().getDeviceAttributes());
-            }
+        if (DeviceAttributesManager.getInstance().getDeviceAttributes() != null
```

Comment:
should we do a null check against the shared instance here, like the one on line 387? if it's null, it seems like it would crash at this point before we ever got to it.